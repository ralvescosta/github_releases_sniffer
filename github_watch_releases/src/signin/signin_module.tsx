import React from 'react';
import {SignInScreen} from './presenter/signin.screen';

import {GithubSearchUserRepository} from './infrastructure/repositories/github.search.user.repository';
import {SingInViewController} from './interfaces/signin.view.controller';
import {LocalDatabaseRepository} from './infrastructure/repositories/local.database.repository';
import {SignInUsecases} from './application/usecases/signin.usecases';
import {CheckUserIsLoggedUsecase} from './application/usecases/check.user.is.logged.usecase';

export const WarperSignInScreen: React.FC = () => {
  const githubFindUserRepository = new GithubSearchUserRepository();
  const localDatabaseRepository = new LocalDatabaseRepository();
  const signInUsecases = new SignInUsecases(githubFindUserRepository, localDatabaseRepository);
  const checkUserIsLoggedUsecase = new CheckUserIsLoggedUsecase(localDatabaseRepository);
  const singInViewController = new SingInViewController(signInUsecases, checkUserIsLoggedUsecase);

  return <SignInScreen viewController={singInViewController} />;
};
