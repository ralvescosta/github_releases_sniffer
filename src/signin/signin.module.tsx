import React from 'react';
import {SignInScreen} from './presenter/signin.screen';

import {GithubSearchUserRepository} from './infrastructure/repositories/github.search.user.repository';
import {SingInViewController} from './interfaces/signin.view.controller';
import {LocalDatabaseRepository} from './infrastructure/repositories/local.database.repository';
import {SignInUsecases} from './application/usecases/signin.usecases';

export const SignInScreenModule = () => {
  const githubSearchUserRepository = new GithubSearchUserRepository();
  const localDatabaseRepository = new LocalDatabaseRepository();
  const signInUsecases = new SignInUsecases(githubSearchUserRepository, localDatabaseRepository);
  const singInViewController = new SingInViewController(signInUsecases);

  return <SignInScreen viewController={singInViewController} />;
};
