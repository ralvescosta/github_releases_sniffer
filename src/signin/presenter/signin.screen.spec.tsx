import React from 'react';
import {cleanup, render, RenderAPI, fireEvent} from '@testing-library/react-native';

import {TextInput, TouchableOpacity, ActivityIndicator} from 'react-native';

import {SignInScreen} from './signin.screen';
import {SingInViewControllerSpy} from './__test__/signin.view.controller.spy';

type SutTypes = {
  sut: RenderAPI;
  singInViewControllerSpy: SingInViewControllerSpy;
};

function makeSut(): SutTypes {
  let singInViewControllerSpy: any;

  const SignInScreenWarper = () => {
    singInViewControllerSpy = new SingInViewControllerSpy();
    return <SignInScreen viewController={singInViewControllerSpy} />;
  };

  const sut = render(<SignInScreenWarper />);

  return {
    sut,
    singInViewControllerSpy,
  };
}

describe('SignIn Screen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it('Should render correctly', () => {
    const {sut} = makeSut();
    expect(sut.toJSON()).toMatchSnapshot();
  });

  it('Should render TextInput and TouchableOpacity', () => {
    const {sut} = makeSut();

    expect(sut.UNSAFE_getByType(TextInput).props.placeholder).toBe('Github user');
    expect(sut.UNSAFE_getByType(TouchableOpacity).props.testID).toBe('signin-button');
  });

  it('Should render ActivityIndicator into a TouchableOpacity if onPressed', () => {
    const {sut} = makeSut();

    sut.UNSAFE_getByType(TouchableOpacity).props.onPress();
    expect(sut.UNSAFE_getByType(ActivityIndicator).props.testID).toBe('signin-indicator');
  });

  it('Should ', () => {
    const {sut, singInViewControllerSpy} = makeSut();

    const textInput = sut.UNSAFE_getByType(TextInput);
    fireEvent.changeText(textInput, 'some_text');

    expect(singInViewControllerSpy.inputValue).toBe('some_text');
  });

  // it('Should render Alert if onPressSignIn throws', () => {
  //   const {sut, singInViewControllerSpy} = makeSut();
  //   singInViewControllerSpy.inputValue = '';

  //   sut.root.findByType(TouchableOpacity).props.onPress();
  // });
});
