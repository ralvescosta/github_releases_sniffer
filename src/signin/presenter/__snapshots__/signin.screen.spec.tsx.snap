// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Screen Should render correctly: SingIn Screen to Default Elements 1`] = `ReactWrapper {}`;
